elements/pf-text-input/pf-text-input.ts

```diff
@@ -0,0 +1,261 @@
+ * @cssprop --pf-c-form-control--textarea--invalid--BackgroundPositionY - {@default var(--pf-c-form-control--PaddingLeft)}
+ * @cssprop --pf-c-form-control--m-icon-sprite--success--BackgroundUrl - {@default url(/v4/images/status-icon-sprite.4fee9fefc3971799d2dd4de0a15617f0.svg#success)}
+ * @cssprop --pf-c-form-control--m-icon-sprite--m-warning--BackgroundUrl - {@default url(/v4/images/status-icon-sprite.4fee9fefc3971799d2dd4de0a15617f0.svg#warning)}
+ * @cssprop --pf-c-form-control--m-icon-sprite--invalid--BackgroundUrl - {@default url(/v4/images/status-icon-sprite.4fee9fefc3971799d2dd4de0a15617f0.svg#invalid)}
+ * @cssprop --pf-c-form-control--m-icon-sprite__select--BackgroundUrl - {@default url(/v4/images/status-icon-sprite.4fee9fefc3971799d2dd4de0a15617f0.svg#select)}
+ * @cssprop --pf-c-form-control--m-icon-sprite--m-search--BackgroundUrl - {@default url(/v4/images/status-icon-sprite.4fee9fefc3971799d2dd4de0a15617f0.svg#search)}
+ * @cssprop --pf-c-form-control--m-icon-sprite--m-calendar--BackgroundUrl - {@default url(/v4/images/status-icon-sprite.4fee9fefc3971799d2dd4de0a15617f0.svg#calendar)}
+ * @cssprop --pf-c-form-control--m-icon-sprite--m-clock--BackgroundUrl - {@default url(/v4/images/status-icon-sprite.4fee9fefc3971799d2dd4de0a15617f0.svg#clock)}
+ * @cssprop --pf-c-form-control--m-icon-sprite__select--BackgroundSize - {@default var(--pf-c-form-control--FontSize)}
+ * @cssprop --pf-c-form-control--m-icon-sprite__select--BackgroundPositionX - {@default calc(100% - var(--pf-global--spacer--md, 1rem) + 7px)}
+ * @cssprop --pf-c-form-control--m-icon-sprite__select--success--BackgroundPosition - {@default calc(100% - var(--pf-global--spacer--md, 1rem) + 1px - var(--pf-global--spacer--lg, 1.5rem))}
+ * @cssprop --pf-c-form-control--m-icon-sprite__select--m-warning--BackgroundPosition - {@default calc(100% - var(--pf-global--spacer--md, 1rem) - var(--pf-global--spacer--lg, 1.5rem) + 0.0625rem)}
+ * @cssprop --pf-c-form-control--m-icon-sprite__select--invalid--BackgroundPosition - {@default calc(100% - var(--pf-global--spacer--md, 1rem) - var(--pf-global--spacer--lg, 1.5rem))}
+ */
+@customElement('pf-text-input')
+export class PfTextInput extends LitElement {
+  static readonly styles = [styles];
+
+  static readonly formAssociated = true;
+
+  static override shadowRootOptions: ShadowRootInit = { ...LitElement.shadowRootOptions, delegatesFocus: true };
+
+  /** Trim text on left */
+  @property({ type: Boolean, reflect: true, attribute: 'left-truncated' }) leftTruncated = false;
+
+  /** Value to indicate if the input is modified to show that validation state.
+   * If set to success, input will be modified to indicate valid state.
+   * If set to warning,  input will be modified to indicate warning state.
+   * Invalid inputs will display an error state
+   */
+  @property({ reflect: true }) validated?: 'success' | 'warning';
```

Comment:
PFv4 looks to have `'success' | 'warning' | 'error' | 'default'` is there a reason not to have the `error` option here?

Reply:
I was thinking if set declaratively it could be set by a server template form rerender?  Maybe I'm not thinking through that completely.  Does this always assume client side validation?

Reply:
it could be :shrug: i'd be willing to defer that until the request comes in